cpp/src/arrow/ipc/reader.cc

```diff
@@ -1061,6 +1062,31 @@ class RecordBatchFileReaderImpl : public RecordBatchFileReader {
     return internal::GetMetadataVersion(footer_->version());
   }
 
+  static Status LoadFieldsSubset(const flatbuf::RecordBatch* metadata,
+                                 const IpcReadOptions& options,
+                                 io::RandomAccessFile* file,
+                                 const std::shared_ptr<Schema>& schema,
+                                 const std::vector<bool>* inclusion_mask,
+                                 MetadataVersion metadata_version = MetadataVersion::V5) {
+    ArrayLoader loader(metadata, metadata_version, options, file);
+    for (int i = 0; i < schema->num_fields(); ++i) {
+      const Field& field = *schema->field(i);
+      if (!inclusion_mask || (*inclusion_mask)[i]) {
+        // Read field
+        auto column = std::make_shared<ArrayData>();
+        RETURN_NOT_OK(loader.Load(&field, column.get()));
```

Comment:
This column is also only used to get the side effect of Load? In that case this ArrayData can be stack-allocated too I think.

Reply:
Good catch. I copied this code snippet from `ArrayLoader`'s usage, and in this case, it is not needed too be a `shared_ptr`. Fixed in the latest commit.